cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -721,25 +1062,56 @@ class PointerArithmeticInstruction extends BinaryInstruction {
 
   final override string getImmediateString() { result = elementSize.toString() }
 
+  /**
+   * Gets the size of the element pointed to by the pointer, in bytes.
```

Comment:
"element(s)", "pointer(s)"?

Reply:
I understand the confusion, but I can't come up with a better way to word it that doesn't suggest that the predicate could return multiple values. Ideas I've considered:

"Gets the size of each element pointed to by the pointer..."
"Gets the size of the element type pointed to by the pointer..."
"Gets the size of the individual elements pointed to by the pointer..."

Maybe something completely different, like:

"Gets the number of bytes between successive elements pointed to by the pointer."

Any other suggestions appreciated.

Reply:
How about explicitly calling out the issue, something like:
```
Gets the size of the element type pointed to by the pointer or pointers, in bytes. If this instruction has more than one pointer operand, they must have the same pointer type, so this predicate can only have one result.
```

Reply:
I've added a detail paragraph to this QL doc, and changed the wording of the summary slightly. I think it gets the right point across now.